rust/lance/src/datafusion/dataframe.rs

```diff
@@ -124,7 +137,10 @@ impl TableProvider for LanceTableProvider {
         scan.limit(limit.map(|l| l as i64), None)?;
         scan.scan_in_order(self.ordered);
 
-        scan.create_plan().await.map_err(DataFusionError::from)
+        let plan = scan.create_plan().await.map_err(DataFusionError::from)?;
+        
+        // Wrap the plan with tracing to emit lance::execution events
+        Ok(Arc::new(TracedLanceExec::new(plan)))
```

Comment:
can we directly leverage `report_plan_summary_metrics`. Can you describe what the exact situation that is not emitting metrics? (maybe describe it as a test so I can try to reproduce and play around?)